### ISSUE TITLE
Relocating shards is only done from single source

### ISSUE THREAD

**Issue by shaharmor** [elastic/elasticsearch#15645]
Hi,

I had 2 servers with 200 indices on them, each index with 5 primaries & 1 replica.
All indices have `"index.routing.allocation.exclude.tag" : "hot"` set on them.

I then added 3 additional servers with tag `hot` to the cluster.
Next i updated the settings on 30 indices to:
```
"index.routing.allocation.include.tag" : "hot",
"index.routing.allocation.exclude.tag" : ""
```

What happened is that all 30 indices started relocating but all from the same source (A single server out of the original two).

Which means that i'm basically using just 50% of my transfer rate.

Any ideas what it might be?

ES version 1.5.2

**Comment by s1monw** [comment]
So my initial assumption is that you are mainly unlucky. We don't throttle outgoing recovery/relocations yet so it's totally possible that they move all from one machine. Do you know if you have any recovery settings set on your cluster? 

https://github.com/elastic/elasticsearch/pull/15372 <== this will help here it won't allow all recoveries to come from the same source

**Comment by shaharmor** [comment]
these are my related settings:
```
indices.recovery.max_bytes_per_sec: "100mb"
indices.recovery.concurrent_streams: 5
cluster.routing.allocation.node_concurrent_recoveries: 5
```

not sure if anything else is related

**Comment by s1monw** [comment]
> cluster.routing.allocation.node_concurrent_recoveries: 5

that limits the target nodes not the source that's why all can come from the same source

> indices.recovery.concurrent_streams: 5

that limits now many concurrent threads we use to send files from the source so all recoveries will shard 5 workers

> indices.recovery.max_bytes_per_sec: "100mb"

that's throttling IO rate of the 5 workers
I don't think here is anything you can do about all this, that's just the way it works :(

**Comment by shaharmor** [comment]
even if i increase the concurrent recoveries and streams to 10, the extra 5 shards after still coming from the same source... looks very "unlucky" to me :(

**Comment by s1monw** [comment]
> even if i increase the concurrent recoveries and streams to 10, the extra 5 shards after still coming from the same source... looks very "unlucky" to me :(

as I said - nothing you can do about this until #15372 is available which will be likely 3.0

**Comment by shaharmor** [comment]
Got it. thanks a lot for the information!
